File: 2015/004/part_one.py
import common
import hashlib

if __name__ == "__main__":
    for input_file in common.inputs:
        for line in common.read_file(input_file):
            to_add = 0
            while True:
                md5 = hashlib.md5(("%s%d" % (line, to_add)).encode())
                digested = md5.hexdigest()
                if digested.startswith("00000"):
                    print(line, to_add, digested)
                    break
                to_add += 1

File: kotlin/src/main/kotlin/pt/dmb/adventofcode/y2016/p05/PartOne.kt
package pt.dmb.adventofcode.y2016.p05

import pt.dmb.adventofcode.common.Utilities

class PartOne {
}

fun main() {
    Utilities.readInputFile(2016, 5, scope = "main").forEach { line ->
        val solver = PartOne()
        TODO()
    }
}

/*
from hashlib import md5

import common

if __name__ == "__main__":
    for input_file in common.inputs:
        print(input_file)
        password = ""

        for line in common.read_file(input_file):
            test_value = 1
            while True:
                test = line + str(test_value)
                test_md5 = md5(test.encode("utf-8")).hexdigest()

                if test_md5.startswith("00000"):
                    password += test_md5[5]
                    if len(password) == 8:
                        print(password)
                        break

                test_value += 1

c6697b55
 */